Deprecate Bbox.anchored() with no container. (#24913)
matplotlib/matplotlib: doc/api/next_api_changes/deprecations/24913-AL.rst

``bbox.anchored()`` with no explicit container
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> If no *container* is passed to `.BboxBase.anchored`, the call simply returns a
> frozen copy of the original bbox; this is now deprecated.

Reviewer: ```suggestion
Not passing a *container* argument to `.BboxBase.anchored` is now deprecated.
```

Author: done

Reviewer: Anyone should merge when green enough....
